src/py/pyodide/http.py

```diff
@@ -202,9 +202,7 @@ async def unpack_archive(self, *, extract_dir=None, format=None):
         """
         buf = await self.buffer()
         filename = self._url.rsplit("/", -1)[-1]
-        unpack_buffer_archive(
-            buf, filename=filename, format=format, extract_dir=extract_dir
-        )
+        unpack_buffer(buf, filename=filename, format=format, extract_dir=extract_dir)
```

Comment:
If we are going to use `unpack_buffer` here too. I think `unpack_buffer` should be more general, than having specific features of handling Python packages.

Reply:
Well it's a private function so I think it can do what it wants. I actually think all of our public apis for unpacking archives should be modified to have an option to compile wasm they encounter. Then they will all have good reason to use this.

My reasoning is that I want to advertise that `pyfetch` or `pyodide.unpackArchive` can be used to install packages with no dependency management. But they can't install packages with any binary extensions because they don't compile wasm. So I think we should make it an option for them to compile wasm (and maybe make it the default option).

Reply:
> Well it's a private function so I think it can do what it wants.

My tiny concern is that we may update `unpack_buffer` and it results in unexpected side effect on `FetchResponse.unpack_archive`.

> I actually think all of our public apis for unpacking archives should be modified to have an option to compile wasm they encounter. Then they will all have good reason to use this.

> My reasoning is that I want to advertise that `pyfetch` or `pyodide.unpackArchive` can be used to install packages with no dependency management. But they can't install packages with binary extensions because they don't compile wasm. So I think we should make it an option for them to compile wasm (and maybe make it the default option).

Yes, I agree that we want and should make our unpacking APIs to be used for installing packages.
I think the problem is that we don't have `urllib` or `requests` support yet, so people should also use them for a general perpose unpacking method.

Reply:
> so people should also use them for a general purpose unpacking method.

Yeah they can use that too. I think we should add a `compile_wasm` boolean argument that controls the behavior. But most `.so` files that people download are probably actual wasm libraries, so I think it would be okay to default true and having people turn it off if they have `.so` files that aren't wasm libraries (or if they are wasm libraries but they don't want to compile them until later for some reason).